src/nodes.coffee

```diff
@@ -885,30 +976,51 @@ exports.StatementLiteral = class StatementLiteral extends Literal
   compileNode: (o) ->
     [@makeCode "#{@tab}#{@value};"]
 
+  astType: ->
+    switch @value
+      when 'continue' then 'ContinueStatement'
+      when 'break'    then 'BreakStatement'
+      when 'debugger' then 'DebuggerStatement'
```

Comment:
I’ve been dealing with an issue where the browser compiler needs to be minified with variable name mangling disabled, to avoid breaking code like `if this instanceof Literal` (because with mangling on, `Literal` gets renamed to something like `a`, and then the `instanceof` check fails). I’ve been considering adding a `type` property to every single node class, so that I could rewrite those `instanceof` checks to be `if this.type is 'Literal'` and therefore the compiler would be safe for variable name mangling.

Reply:
@GeoffreyBooth I don't particularly understand the implications of minification but I'm surprised that particular example would break, since `Literal` is a reference that presumably should be preserved (regardless of whether it's being used in an `instanceof` check), seems like where you could get into trouble would be relying on the string `'Literal'` as the class name

As far as it relates to this PR, I wonder if name mangling would affect using `@constructor.name` as the default AST `type`? If you were to make that change to allow name mangling, then perhaps that would need to be updated so that on `Base` it'd be `astType: -> @type` instead of `astType: -> @constructor.name`